### PR TITLE
Feature/make auth ready

### PR DESCRIPTION
**Pull request by grandyzzz**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#PUBLIC_APP_ENV=prod
```
Comment (QDieu) [inline .env]: This doesn't work, the variable name must begin REACT_APP_VARIABLE_NAME

File: src/components/RequireAuth/RequireAuth.js [hunk 2]
```diff
@@ -1,8 +1,49 @@
...
+        } = JSON.parse(credentials);
+        try {
+          const user = await getUserById(public_id, token);
+          if (user.confirmed) {
+            dispatch(setUser({
+              isAuth: true,
+            }));
+          }
+        } catch (err) {
+          console.log('error getting user', err.message);
```
Comment (QDieu) [inline src/components/RequireAuth/RequireAuth.js]: console.error better suited for error output

File: src/components/SignUpScreen/SignUpScreen.js [hunk 3]
```diff
@@ -30,42 +30,46 @@ export default function SignUp() {
...
-              navigate('../login');
-            }
-          })
-          .catch((err) => console.log(err)); // todo add logic
-      } else {
-        console.log('passwords are not the same'); // todo add logic
-      }
-    } else {
-      console.log('not valid fields'); // todo add logic
+    if (!isPasswordEqual) {
```
Comment (QDieu) [inline src/components/SignUpScreen/SignUpScreen.js]: Might be better now (password === retypePassword)

File: src/slices/authSlice.js [hunk 4]
```diff
@@ -31,3 +31,4 @@ export const selectUserCredentials = state => {
     authToken: state.auth.user.auth_token
   };
 };
+export const selectIsAuth = state => state.auth.user.isAuth;
```
Comment (QDieu) [inline src/slices/authSlice.js:33]: Need to put it in a file selectors.js (index maybe) 
isAuth = useSelector() and export isAuth

File: .env (hunk 1 above)
Comment (Ali-Gator) [reply .env]: Is there any prove that this doesn't work? Any way as I see it's renamed in redux-toolkit branch

Comment (grandyzzz) [reply .env]: @Ali-Gator https://create-react-app.dev/docs/adding-custom-environment-variables/

File: src/components/RequireAuth/RequireAuth.js (hunk 2 above)
Comment (Ali-Gator) [reply src/components/RequireAuth/RequireAuth.js]: It's temporary. Next it should be replace with toasts.

File: src/components/SignUpScreen/SignUpScreen.js (hunk 3 above)
Comment (Ali-Gator) [reply src/components/SignUpScreen/SignUpScreen.js]: password !== retypePassword added

File: src/slices/authSlice.js (hunk 4 above)
Comment (Ali-Gator) [reply src/slices/authSlice.js:33]: I like concept that all things considering state slices pass in one file. Here best practice from redux itself:
https://redux.js.org/tutorials/essentials/part-5-async-logic

image

**Comment by Ali-Gator** [comment]
All comments has resolved. Sent one commit to redux-toolkit branch.

**Pull request merged**